addons/web/static/src/core/signature/name_and_signature.js

```diff
@@ -84,6 +84,10 @@ export class NameAndSignature extends Component {
                     if (this.state.signMode === "auto") {
                         this.drawCurrentName();
                     }
+                    if (this.props.signature.signatureImage) {
+                        this.clear();
+                        this.fromDataURL(this.props.signature.signatureImage);
+                    }
```

Comment:
Just to be sure there are no confusion : if the `signMode` is `auto` *and* you provide a default signature, the last one will override the automatically generated one, right ?

Reply:
When you open the dialog, it is in the "draw" mode by default. this.drawCurrentName() is not called and the existing signature image is loaded. If you click on "Auto", the drawed signature is reset and we replace it with the result of this.drawCurrentName();
That's why I made this change: https://github.com/odoo/enterprise/pull/64685/files#diff-6b95c8b9c96f6a2f70666325c1e34b7253fd8bcd301a6848f0971d67a5a32929R140-R142

As you commented, this is not great because we modify the props inside a getter but I couln' find a better way. I could refine the condition here with this to avoid confusion:

```js
if (this.state.signMode === "draw" && this.props.signature.signatureImage) {
    this.clear();
    this.fromDataURL(this.props.signature.signatureImage);
}
```